### DISCUSSION TITLE
Should global error catch the root layout error?

### DISCUSSION THREAD

**Discussion by RafalChojnowskiPagepro** [vercel/next.js#52471] [App Router]
I'm wondering if I understood global-error.tsx idea correctly. 
Here's the code sandbox which for me should work: https://codesandbox.io/p/sandbox/relaxed-glade-yfstnr 
But it doesn't catch the error thrown in root layout (for me it should). 
Could somebody explain it to me? 🙏

**Comment by icyJoseph** [comment]
Hi,

You are correct, but the testing method you are using is not. Since you throw 100% of the time, you won't see anything meaningful.

```tsx

export default function RootLayout({ children }) {
  if (Math.random() > 0.5) {
    throw new Error("Root error");
  }

  return (
    <html>
      <head />
      <body>{children}</body>
    </html>
  );
}

```

And then you make your page:

```tsx
export const dynamic = "force-dynamic";

export default function Home() {
  return <p>Home sweet home</p>;
}

```

Now build and run the app. If you go to `/`, it'll work sometimes, but other times, your global error boundary will kick in.

**Reply by RafalChojnowskiPagepro** [reply]
I've edited sandbox as per your reply and global error boundary is still missing

**Reply by icyJoseph** [reply]
Yeah in dev mode, throwing errors like that, won't work I think, you need to build the app. Notice I never mentioned dev mode.
